addons/point_of_sale/static/src/js/Screens/ClientListScreen/ClientListScreen.js

```diff
@@ -150,15 +172,36 @@ odoo.define('point_of_sale.ClientListScreen', function(require) {
             };
             this.render();
         }
-        async saveChanges(event) {
+        async saveChanges() {
             try {
+                let processedChanges = {};
+                for (let [key, value] of Object.entries(this.changes)) {
+                    if (this.intFields.includes(key)) {
+                        processedChanges[key] = parseInt(value) || false;
+                    } else {
+                        processedChanges[key] = value;
+                    }
+                }
+                if ((!this.state.editModeProps.partner.name && !processedChanges.name) ||
+                    processedChanges.name === '') {
+                    return this.showPopup('ErrorPopup', {
+                        title: _t('A Customer Name Is Required'),
+                    });
+                }
+                processedChanges.id = this.state.editModeProps.partner.id || false;
+
                 let partnerId = await this.rpc({
                     model: 'res.partner',
                     method: 'create_from_ui',
-                    args: [event.detail.processedChanges],
+                    args: [processedChanges],
                 });
+                this.changes = {};
```

Comment:
Can you put this in a `finally` clause? Because regardless of whether the save failed or not, the changes should be reset.

Reply:
done!